cmd/sops/main.go

```diff
@@ -2262,7 +2266,9 @@ func toExitError(err error) error {
 
 func keyservices(c *cli.Context) (svcs []keyservice.KeyServiceClient) {
 	if c.Bool("enable-local-keyservice") {
-		svcs = append(svcs, keyservice.NewLocalClient())
+		// propagate azure-kv-skip-uri-validation flag to local keyservice server instance
+		local := keyservice.NewCustomLocalClient(keyservice.Server{Prompt: false, SkipAzureKvUriValidation: c.Bool("azure-kv-skip-uri-validation")})
```

Comment:
You probably have to use both `c.Bool()` and `c.GlobalBool`, since `c.Bool` will likely not provide the right value in subcommands.